setup.py

```diff
@@ -42,6 +42,12 @@
     "partd >= 1.2.0",
     "pyyaml >= 5.3.1",
     "toolz >= 0.8.2",
+    # importlib.metadata has the following bugs fixed in 3.10.9 and 3.11.1
+    # https://github.com/python/cpython/issues/99130
+    # https://github.com/python/cpython/issues/98706
+    # TODO: when python 3.12 is support is added this should be a
+    # conditional dependency
+    "importlib_metadata >= 4.13.0",
```

Comment:
I originally had this as:

```python
"importlib_metadata >= 4.13.0; python_full_version < '3.10.9' or ('3.11a0' <= python_full_version < '3.11.0')"
```
and then in the code:

```python
if sys.version_info < (3, 10, 9) or ((3, 11) <= sys.version_info < (3, 11, 1)):
    from importlib_metadata import entry_points
else:
    from importlib.metadata import entry_points
```

however the conda recipe only supports conditional dependencies on major versions eg: ` - importlib_metadata # py<312` and it made the code a bit complicated on every import, and we don't support 3.12 yet so I made it a mandatory release with a TODO: to make it conditional when we add 3.12 support.

Reply:
We could probably use Jinja in the Conda recipe to add a more subtle condition, but that might not be worth it if we already prefer `importlib_metadata` for other reasons